src/drive/web/modules/actionmenu/ActionsItems.jsx

```diff
@@ -0,0 +1,50 @@
+import React from 'react'
+import cx from 'classnames'
+import { useI18n } from 'cozy-ui/transpiled/react/I18n'
+
+import MenuItem from './MenuItem'
+import styles from 'drive/styles/actionmenu.styl'
+
+export const ActionsItems = ({ actions, file, onClose }) => {
+  const { t } = useI18n()
+  let previousAction = ''
+  return actions.map((actionObject, i) => {
+    const actionName = Object.keys(actionObject)[0]
+    const actionDefinition = Object.values(actionObject)[0]
+    if (
+      actionDefinition.displayCondition !== undefined &&
+      !actionDefinition.displayCondition([file])
+    )
+      return null
+    if (previousAction === actionName) {
+      previousAction = actionName
+      return null
+    } else {
+      previousAction = actionName
+    }
```

Comment:
This feels a bit weird, shouldn't the filtering be done before ? Why do we have to do this (returning null if the previous action is the same) ?

Also can you add a docstring ?

Reply:
I wonder why the filtering is not done prior to rendering ? Why do we need to do it at each render ? I feel like the filtering/duplicate-removing code should be outside the component.

Reply:
done in 15d1906